gpustack/detectors/nvidia_smi/nvidia_smi.py

```diff
@@ -104,8 +106,12 @@ def _run_command(self, command):
 
     def _command_gather_gpu(self):
         executable_command = [
-            "nvidia-smi",
+            self.exec_command,
             "--format=csv,noheader",
             "--query-gpu=index,name,memory.total,memory.used,utilization.gpu,temperature.gpu",
         ]
         return executable_command
+
+
+class WSLNvidiaSMI(NvidiaSMI):
```

Comment:
I prefer we hide the implementation detail in NvidiaSMI so that the caller don't need extra knowledge. For example, iterate possible command values and use anything that is available.